Alloy/lib/alloy.js

```diff
@@ -417,7 +417,18 @@ exports.createWidget = function(id, name, args) {
  * @return {Alloy.Controller} Alloy controller object.
  */
 exports.createController = function(name, args) {
-	return new (require('alloy/controllers/' + name))(args);
+	try {
+		return new (require('alloy/controllers/' + name.replace(/^\//, '')))(args);
+	} catch (err) {
+		if(OS_IOS && err.indexOf('architecture:') !== -1) {
```

Comment:
Right now, we are adding a hack in Alloy code to address only iOS specific errors. If there is something wrong with Android platform sooner, we can't keep on adding multiple condition checks based on the different errors from the platform.

I'm not a fan of hacking Alloy code to show relevant error messages. We should figure out the changes that triggered this problem in SDK (or where ever) and try to address the root cause, rather than adding a hack in Alloy code.